processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -1349,10 +1349,12 @@ boolean needConversion(int index)
     {
       IntBuffer readOnly = conversions[index].asReadOnlyBuffer();
       readOnly.rewind();
-      for (int i = 0; readOnly.hasRemaining(); i++) {
+      int i = 0;
+      while (readOnly.hasRemaining()) {
```

Comment:
There is an inspection that checks that loop variable is checked in condition part of for loop. This occasion was a false positive, and I think it's better to reformat code than add `@SuppressWarnings`